pp_netlib/network.py

```diff
@@ -137,38 +141,51 @@ def construct(self, network_data, weights = None): #, previous_network = None, a
         else:
             vertex_labels = self.ref_list
             self_comparison = True
-
+        self.vertex_labels = vertex_labels
+        self.weights = weights
         # initialise a graph object
         if self.backend == "GT":
-            self.graph = construct_with_graphtool(network_data=network_data, vertex_labels=vertex_labels, weights=weights)
+            self.graph = construct_with_graphtool(network_data=network_data, vertex_labels=self.vertex_labels, weights=weights)
         elif self.backend == "NX":
-            self.graph = construct_with_networkx(network_data=network_data, vertex_labels=vertex_labels, weights=weights)
+            self.graph = construct_with_networkx(network_data=network_data, vertex_labels=self.vertex_labels, weights=weights)
 
-        ## keeping this section here for now; might be useful in add_to_network method
-        # ########################
-        # ####  PREV NETWORK  ####
-        # ########################
-        # if previous_network is not None:
-        #     prev_edges = []
-        #     if weights is not None:
-        #         extra_sources, extra_targets, extra_weights = process_previous_network(previous_network = previous_network, adding_qq_dists = adding_qq_dists, old_ids = old_ids, previous_pkl = previous_pkl, vertex_labels = vertex_labels, weights = (weights is not None), use_gpu = use_gpu)
-        #         for (src, dest, weight) in zip(extra_sources, extra_targets, extra_weights):
-        #                 prev_edges.append((src, dest, weight))
 
-        #     else:
-        #         extra_sources, extra_targets = process_previous_network(previous_network = previous_network, adding_qq_dists = adding_qq_dists, old_ids = old_ids, previous_pkl = previous_pkl, vertex_labels = vertex_labels, weights = (weights is not None), use_gpu = use_gpu)
-        #         for (src, dest, weight) in zip(extra_sources, extra_targets):
-        #                 prev_edges.append((src, dest))
+    def prune(self, type_isolate = None):
 
+        if self.graph is None:
+            raise RuntimeError("Graph not constructed or loaded.")
 
-        #     self.graph.add_edge_list(prev_edges) ## add previous edge list to newly made graph
+        if self.backend == "GT":
+            reference_vertices = set()
+            components = self.gt.label_components(self.graph)[0].a
 
-        #     self.edges = [edge for edge in self.graph.edges()]
+            for component in set(components):
+                reference_indices = gt_clique_prune(component, self.graph, set(), components)
+                reference_vertices.update(reference_indices)
 
+            labels = list(self.graph.vp["id"][v] for v in self.graph.vertices())
+            self.graph = gt_get_ref_graph(self.graph, reference_vertices, labels, type_isolate, backend=self.backend)
```

Comment:
Are you sure it's best to overwrite `self.graph`? I think either storing as `self.ref_graph` would probably be better. Or perhaps `prune` should return another `Graph` object?

Reply:
Also I think this function has the side effect of writing to file – it would be good to note that in a comment

Reply:
storing pruned graph as `self.ref_graph`, the function does not save pruned graph to file, should I add that?

Reply:
Yes, I think a sensible side-effect of running prune is that the pruned graph is saved to file. Alternatively, you could add a `pruned` argument to `save`, which saves the pruned graph `self.ref_graph` instead of `self.graph`

Reply:
modified save method as follows: moved save logic into pp_netlib.functions.save_graph; added extra arg `to_save` to save method so user can specify whether to save "full_graph", "pruned_graph", or "both"; also handled case where pruned graph is not found but user specifies "both": in this case, warning is printed and only full graph is saved